lib/recipe_repository.rb

```diff
@@ -15,4 +16,10 @@ def all_recipes
       local_source.all_recipes
     end
   end
+
+  def recipes_with(ingredients:)
+    all_recipes.select do |recipe|
+      ingredients.all? { |ingredient| recipe.contains?(ingredient) }
+    end
```

Comment:
Nice 👍  I like the split between the repository and using methods on the domain object.